plugins/generator-1.18.2/forge-1.18.2/utils/particles.java.ftl

```diff
@@ -17,7 +17,7 @@ for(int l=0;l< ${amount}; ++l) {
 	double x0 = x + 0.5 + (random.nextFloat() - 0.5) * ${radius}D * 20;
 	double y0 = y + 1.2 + (random.nextFloat() - 0.5) * ${radius}D;
 	double z0 = z + 0.5 + (random.nextFloat() - 0.5) * ${radius}D * 20;
-	world.addParticle(${particleObj.toString()}, x0, y0, z0, 0, 0, 0);
+	world.addParticle(${particleObj.toString()}, x0-0.5, y0, z0-0.5, 0, 0, 0);

```

Comment:
0.5 is added in statements for x0 and z0 so deducting here is not optimal.

Also, this will make block particles generate in corners, so a universal solution is needed. Applies to all changes.